### DISCUSSION TITLE
[Suggestion] Backward if/when expression

### DISCUSSION THREAD

**Discussion by surfsky** [dotnet/csharplang#3494] [Language Ideas]
Just like python syntax, we can use tail-if expression to kill one line code.

``` csharp
n = 0 if (b == true);  
or
n = 0 when (b == true);
```
means

``` csharp
if (b == true)
   n = 0;
```

**Comment by svick** [comment]
There have been quite a lot of a similar proposals, see https://github.com/dotnet/csharplang/issues/724, https://github.com/dotnet/csharplang/issues/2352#issuecomment-474407239 and https://github.com/dotnet/csharplang/issues/724#ref-issue-266713237.

**Comment by Unknown6656** [comment]
What's speaking against the "classic" ternary if?
```csharp
n = b ? 0 : n;
```
It is shorter, and one does not have to introduce a new syntax.

**Comment by DavidArno** [comment]
So be clear, you are asking that the language support:
```cs
n = 0 if (b);
```
as well as:
```cs
if (b) n = 0;
```
"*to kill one line code*"?

**Comment by surfsky** [comment]
@DavidArno 
Usually, we write if (condition) with a newline character to obey program specs.

``` csharp
if (condition)
    someaction();
```
change to inline expression:

``` csharp
someaction() if (condition);
or
someaction() when(condition);
```

Just like oral english, you can move **if** to front or back freely.
This inline tail-condition expression  was adopted by many program languages. 
It's just a suggestion.

**Comment by DavidArno** [comment]
It seems a lot easier to me to change those "program specs" than add new syntax to the language. 🤷

**Comment by Thaina** [comment]
I don't agree with syntax but I agree with the idea that we should have syntax for one line `if`/`else`

But I would prefer `?:`

In your case

```
a == true ? DoSomeThing():;
b == false ?: DoOtherThing();
a ^ b ? DoLeft() : DoRight();
```
